### DISCUSSION TITLE
No Wifi stations found

### DISCUSSION THREAD

**Discussion by vronp** [RaspAP/raspap-webgui#877] [Q&A]
Hi all,

I installed RaspAP on a 4B without any apparent issues and it's been working very well as a hotspot.  If my understanding is correct, I should be able to scan for other hotspots under the WiFi client menu item.  Doing so yields no other stations which doesn't make sense for this environment.  There should be 3-4 shown in the list.

I stopped the hotspot via RaspAP and attempted another WiFi client scan with the same result.

I sense I am missing something fundamental but I just can't figure out what that might be.  I would welcome any ideas for troubleshooting this.

thanks!

**Comment by billz** [comment]
There's a FAQ for this https://docs.raspap.com/faq/#scanning

**Reply by vronp** [reply]
Ok, so, the fundamental problem appears to be wpa_supplicant does not run, or stay running, after reboot.  The following command run manually after reboot fixes the problem, in other words, it fixes scanning:

sudo wpa_supplicant -B -Dnl80211,wext -c/etc/wpa_supplicant/wpa_supplicant.conf -iwlan0

By the way, after boot, hostapd always appears to be healthy and I have a hotspot that I can connect to.  This aspect has been rock solid.

The syslog file provides plenty of clues it seems.  However, I have checked and don't have another copy of wpa_supplicant running.  Also, there is no control socket present in /var/run after boot.  Scratching my head at this point....

Apr 25 12:48:20 raspberrypi systemd[1]: Condition check resulted in Copy user wpa_supplicant.conf being skipped.
Apr 25 12:48:21 raspberrypi dhcpcd-run-hooks[481]: wlan0: starting wpa_supplicant
Apr 25 12:48:21 raspberrypi wpa_supplicant[533]: Successfully initialized wpa_supplicant
Apr 25 12:48:21 raspberrypi wpa_supplicant[533]: nl80211: kernel reports: Match already configured
Apr 25 12:48:21 raspberrypi wpa_supplicant[533]: nl80211: kernel reports: Match already configured
Apr 25 12:48:21 raspberrypi wpa_supplicant[533]: nl80211: kernel reports: Match already configured
Apr 25 12:48:21 raspberrypi wpa_supplicant[533]: nl80211: kernel reports: Match already configured
Apr 25 12:48:21 raspberrypi wpa_supplicant[533]: nl80211: kernel reports: Match already configured
Apr 25 12:48:21 raspberrypi wpa_supplicant[533]: nl80211: kernel reports: Match already configured
Apr 25 12:48:21 raspberrypi wpa_supplicant[533]: nl80211: kernel reports: Match already configured
Apr 25 12:48:21 raspberrypi wpa_supplicant[533]: nl80211: kernel reports: Match already configured
Apr 25 12:48:21 raspberrypi wpa_supplicant[533]: nl80211: kernel reports: Match already configured
Apr 25 12:48:21 raspberrypi wpa_supplicant[533]: nl80211: kernel reports: Match already configured
Apr 25 12:48:21 raspberrypi wpa_supplicant[533]: nl80211: kernel reports: Match already configured
Apr 25 12:48:21 raspberrypi wpa_supplicant[533]: nl80211: kernel reports: Match already configured
Apr 25 12:48:21 raspberrypi wpa_supplicant[533]: nl80211: kernel reports: Match already configured
Apr 25 12:48:21 raspberrypi wpa_supplicant[533]: nl80211: kernel reports: Match already configured
Apr 25 12:48:21 raspberrypi wpa_supplicant[533]: nl80211: kernel reports: Match already configured
Apr 25 12:48:21 raspberrypi wpa_supplicant[533]: nl80211: kernel reports: Match already configured
Apr 25 12:48:21 raspberrypi wpa_supplicant[533]: nl80211: kernel reports: Match already configured
Apr 25 12:48:21 raspberrypi wpa_supplicant[533]: nl80211: kernel reports: Match already configured
Apr 25 12:48:21 raspberrypi wpa_supplicant[533]: nl80211: kernel reports: Match already configured
Apr 25 12:48:21 raspberrypi wpa_supplicant[533]: nl80211: kernel reports: Match already configured
Apr 25 12:48:21 raspberrypi wpa_supplicant[533]: nl80211: kernel reports: Match already configured
Apr 25 12:48:21 raspberrypi wpa_supplicant[533]: nl80211: kernel reports: Match already configured
Apr 25 12:48:21 raspberrypi wpa_supplicant[533]: nl80211: kernel reports: Match already configured
Apr 25 12:48:21 raspberrypi wpa_supplicant[533]: ctrl_iface exists and seems to be in use - cannot override it
Apr 25 12:48:21 raspberrypi wpa_supplicant[533]: Delete '/var/run/wpa_supplicant/wlan0' manually if it is not used anymore
Apr 25 12:48:21 raspberrypi wpa_supplicant[533]: Failed to initialize control interface 'DIR=/var/run/wpa_supplicant GROUP=netdev'.#012You may have another wpa_supplicant process already running or the file was#012left by an unclean termination of wpa_supplicant in which case you will need#012to manually remove this file before starting wpa_supplicant again.
Apr 25 12:48:21 raspberrypi wpa_supplicant[533]: nl80211: deinit ifname=wlan0 disabled_11b_rates=0
Apr 25 12:48:21 raspberrypi ifup[422]: /etc/network/if-pre-up.d/wpasupplicant: 120: /etc/network/if-pre-up.d/wpasupplicant: cannot create /dev/stderr: No such device or address
Apr 25 12:48:21 raspberrypi ifup[422]: run-parts: /etc/network/if-pre-up.d/wpasupplicant exited with return code 1

**Reply by vronp** [reply]
So, I'm starting to suspect another service is also involved here but I can't put my finger on it.

**Reply by billz** [reply]
Use `sudo systemctl list-units --type=service --state=running ` and look for any services that could be competing for control of wpa_supplicant.

**Reply by vronp** [reply]
Yes, I had already begun investigating that and my list is hostapd.service, dnsmasq.service, dhcpcd.service, and finally networking.service which happens to show as "failed" in that list.

Is there any way to get more verbose logging during the boot process to determine what might be calling wpa_supplicant?  

I'm also looking at options for the systemd-analyze command to see if that might help.

**Reply by billz** [reply]
> Is there any way to get more verbose logging during the boot process

`journalctl -xe` and `dmesg`. run wpa_supplicant in debug mode with:
```
sudo wpa_supplicant -dd -iwlan0 -c/etc/wpa_supplicant/wpa_supplicant.conf
```
substituting wlan0 for your interface. 

If the message `delete '/var/run/wpa_supplicant/wlan0' manually` persists, deleting this file and executing the above again usually works (assuming you have a valid config).

**Reply by vronp** [reply]
Thanks, I'll try this.
Just for kicks, I spun up a virgin Raspberry Pi instance with the Desktop (not lite) version.....same issue.  A lot less errors in syslog but I still have to reinitialize that socket in order to scan.

**Reply by billz** [reply]
Unable to repo on a clean install. I suspect you may have an outdated kernel version and/or an invalid wpa_supplicant conf. Did you follow the project prerequisites? If you want fewer headaches, be sure to do the former... or switch to Armbian as it exhibits none of these problems.

**Reply by vronp** [reply]
The only deviation was not using the LITE version.  I used the Desktop version, dated 3/4/21, with no modifications or additions.

https://www.raspberrypi.org/software/operating-systems/

**Reply by billz** [reply]
There's a good reason why RPi OS Lite is mentioned here https://docs.raspap.com/#quick-start
